### DISCUSSION TITLE
Create resouce by the exact name

### DISCUSSION THREAD

**Discussion by diegoavitorino** [crossplane/crossplane#5532] [General]
Hi, I'm new on Crossplane, I'm trying to setup a simple aws bucket, I have created composition, definition and custom api.

But my bucket create with name managed-s3-r557m (as a pod), I need bucket name be the exatly name that I specified, whithout "r557m" or the other aleatory characters?

Follow attached from my examples,I don't know if I made a mistake
[crossplane.docx](https://github.com/crossplane/crossplane/files/14823291/crossplane.docx)

**Comment by bobh66** [comment]
You can set the `crossplane.io/external-name` annotation to control the name of the resource in the provider.  https://docs.crossplane.io/latest/concepts/managed-resources/#annotations

Just be careful to ensure that the name is unique because if it's not the existing resource will be imported instead of creating a new resource:  https://docs.crossplane.io/knowledge-base/guides/import-existing-resources/#add-the-external-name-annotation

**Comment by diegoavitorino** [comment]
This annotation crossplane.io/external-name I  put the in the composition, definition or customapi?

**Reply by bobh66** [reply]
It needs to be in the Managed Resource definition created by the Composition, typically via patching or using function-go-templating.  If you are using claims then to ensure uniqueness you could use a combination of the `crossplane.io/claim-name` and `crossplane.io/claim-namespace` labels, for example.  You typically don't want to hard-code this value unless you can be sure that there is only ever going to be one composite instance, because the second instance would import the one created by the first instance, since it has the same name, instead of creating a new one.  You would end up with two composite instances managing the same external MR.

**Comment by diegoavitorino** [comment]
One last thing, how I update changes that I made in composite (a field, or tag, any)
I tried kubectl apply -f composition.yaml but nothing happens

**Reply by bobh66** [reply]
Updates to a `Composition` will create a new `CompositionRevision`.  There is a policy that determines whether these changes get automatically applied to existing Composites or whether the Composites have to be manually updated to use the new CompositionRevision.  See https://docs.crossplane.io/knowledge-base/guides/composition-revisions/#using-composition-revisions

`kubectl describe` of your composite instance will tell you whether the policy is `Manual` or `Automatic`.  If it is `Automatic` and the `compositionRevisionRef` matches the latest version of the `Composition` then the change may not have been applied because it may require Server Side Apply which is only used in Composition Functions and not in traditional Patch and Transform.  Also if the MR is deployed already I don't know what impact, if any, changing the external-name annotation will have.  I don't think it will destroy and recreate the external resource, but it might.
